operator/src/main/java/oracle/kubernetes/operator/PodWatcher.java

```diff
@@ -322,6 +334,66 @@ V1ObjectMeta getMetadata(V1Pod pod) {
     Step createReadAsyncStep(String name, String namespace, String domainUid, ResponseStep<V1Pod> responseStep) {
       return new CallBuilder().readPodAsync(name, namespace, domainUid, responseStep);
     }
+
+    protected DefaultResponseStep<V1Pod> resumeIfReady(Callback callback) {
+      return new DefaultResponseStep<>(null) {
+        @Override
+        public NextAction onSuccess(Packet packet, CallResponse<V1Pod> callResponse) {
+
+          DomainPresenceInfo info = packet.getSpi(DomainPresenceInfo.class);
+          String serverName = (String)packet.get(SERVER_NAME);
+          if ((info != null) && (callResponse != null)) {
+            Optional.ofNullable(callResponse.getResult()).ifPresent(result ->
+                    info.setServerPodFromEvent(getPodLabel(result), result));
+            if (onReadNotFoundForCachedResource(getServerPod(info, serverName), isNotFoundOnRead(callResponse))) {
+              LOGGER.fine(EXECUTE_MAKE_RIGHT_DOMAIN, serverName, callback.getRecheckCount());
+              return doNext(new CallBuilder().readDomainAsync(info.getDomainUid(),
+                      info.getNamespace(), new MakeRightDomainStep(callback,null)), packet);
+            }
+          }
+
+          if (isReady(callResponse.getResult()) || callback.didResumeFiber()) {
```

Comment:
Do we need to reset the recheckCount here?

Reply:
I moved the `recheckCount` variable from DomainPresenceInfo to the Callback object since we create a separate callback instance for each resource. The WaitForReady step registers a new Callback instance in the `resumeWhenReady` method and the previous callback will be removed once the fiber is resumed. With this approach, there's no need to reset the recheckCount since previous callback instance will be garbage collected once the resource is ready.